### PR TITLE
Add tutorial part about custom serialization

### PR DESCRIPTION
**Pull request by charlesbvll**
#### What does this implement/fix? Explain your changes.

The part 4 of the tutorial had an unfinished step 3 about serialization. This PR implements an custom serialization example.

### CODE REVIEW

File: doc/source/tutorial/Flower-4-Client-and-NumPyClient-PyTorch.ipynb [hunk 1]
```diff
@@ -466,36 +466,394 @@
...
+        "    def evaluate(self, ins: EvaluateIns) -> EvaluateRes:\n",
+        "        print(f\"[Client {self.cid}] evaluate, config: {config}\")\n",
+        "\n",
+        "        # Deserialize parameters to NumPy ndarray's using our custom function\n",
+        "        parameters_original = ins.parameters\n",
+        "        ndarrays_original = sparse_parameters_to_ndarrays(parameters_original)\n",
+        "\n",
+        "        set_parameters(self.net, ndarrays_original)\n",
+        "        loss, accuracy = test(self.net, self.valloader)\n",
+        "        # return float(loss), len(self.valloader), {\"accuracy\": float(accuracy)}\n",
```
Comment (danieljanes) [inline doc/source/tutorial/Flower-4-Client-and-NumPyClient-PyTorch.ipynb:637]: This commented-out line can be removed

**Review by danieljanes — CHANGES_REQUESTED** [review]
Looks great, good work!

I left one tiny comment, then we can merge it.

Also, unrelated to this PR, I noticed that on line 894, it shows the wrong file name (which shows up on Google Colab). Can we change this to the actual file name?

**Comment by charlesbvll** [comment]
> Looks great, good work!
> 
> I left one tiny comment, then we can merge it.
> 
> Also, unrelated to this PR, I noticed that on line 894, it shows the wrong file name (which shows up on Google Colab). Can we change this to the actual file name?

Sure, this is done!

**Review by danieljanes — APPROVED** [review]
🚀

**Pull request merged**